sources/packages/web/src/components/common/students/StudentProfile.vue

```diff
@@ -138,6 +171,53 @@ export default defineComponent({
       address.value = studentDetail.value.contact.address;
     };
 
+    const editStudentProfile = async () => {
+      await studentEditProfile.value.showModal(
+        {
+          givenNames: studentDetail.value.firstName,
+          lastName: studentDetail.value.lastName,
+          birthdate: studentDetail.value.dateOfBirth,
+          email: studentDetail.value.email,
+        },
+        saveUpdatedProfileInfo,
+      );
+    };
+
+    const saveUpdatedProfileInfo = async (
+      modalData: UpdateStudentDetailsAPIInDTO | boolean,
+    ) => {
+      try {
+        await StudentService.shared.updateStudentProfileInfo(
+          props.studentId,
+          modalData as UpdateStudentDetailsAPIInDTO,
+        );
+        await loadStudentProfile();
+        snackBar.success("Profile Information updated successfully.");
+        return true;
+      } catch (error) {
+        snackBar.error(
+          "An error happened while updating the profile information. Your profile information could not be updated.",
+        );
+        return false;
+      }
+    };
+
+    const canEditProfile = computed(() => {
+      return (
+        studentDetail.value.identityProviderType ===
+        IdentityProviders.BCeIDBasic
+      );
+    });
+
+    watchEffect(() => {
+      updatedStudentDetails.value = {
+        givenNames: studentDetail.value.firstName,
+        lastName: studentDetail.value.lastName,
+        birthdate: studentDetail.value.dateOfBirth,
+        email: studentDetail.value.email,
+      };
+    });
```

Comment:
This can be removed with the suggested change: https://github.com/bcgov/SIMS/pull/3770/files#r1792156218